actors/builtin/market/deal.go

```diff
@@ -12,6 +24,184 @@ import (
 //}
 var PieceCIDPrefix = market0.PieceCIDPrefix
 
+// The DealLabel is a kinded union of string or byte slice.
+// It serializes to a CBOR string or CBOR byte string depending on which form it takes.
+// The empty label is serialized as an empty CBOR string (maj type 3).
+type DealLabel struct {
+	s  *string
+	bs *[]byte
+}
+
+// Zero value of DealLabel is canonical EmptyDealLabel but three different values serialize the same way:
+// DealLabel{nil, nil}, (*DealLabel)(nil), and DealLabel{&"", nil}
+var EmptyDealLabel = DealLabel{s: nil, bs: nil}
+
+func NewLabelFromString(s string) (DealLabel, error) {
+	if len(s) > DealMaxLabelSize {
+		return EmptyDealLabel, xerrors.Errorf("provided string is too large to be a label (%d), max length (%d)", len(s), DealMaxLabelSize)
+	}
+	if !utf8.ValidString(s) {
+		return EmptyDealLabel, xerrors.Errorf("provided string is invalid utf8")
+	}
+	return DealLabel{
+		s: &s,
+	}, nil
+}
+
+func NewLabelFromBytes(b []byte) (DealLabel, error) {
+	if len(b) > DealMaxLabelSize {
+		return EmptyDealLabel, xerrors.Errorf("provided bytes are too large to be a label (%d), max length (%d)", len(b), DealMaxLabelSize)
+	}
+
+	return DealLabel{
+		bs: &b,
+	}, nil
+}
+
+// True iff neither bytes nor string is set
+func (label DealLabel) IsEmpty() bool {
+	return label.s == nil && label.bs == nil
+}
```

Comment:
Is this accurate? I'd assume that `IsEmpty` would mean "is empty". I.e., `""` would also be empty.